tests/test_cases/issue_1279/issue_1279.py

```diff
@@ -0,0 +1,13 @@
+import cocotb
+
+
+@cocotb.test(expect_error=cocotb.result.SimFailure)
+async def test_sim_failure_a(dut):
+    # invoke a deadlock
+    await cocotb.triggers.RisingEdge(dut.clk)
+
+
+@cocotb.test()
+async def test_sim_failure_b(dut):
+    raise cocotb.result.TestFailure("This test should never run")
```

Comment:
Unless your plan is to wait until after we drop python 2, you can't use `async` here yet.